src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,51 @@
+package duke;
+
+import duke.command.Command;
+
+/**
+ * The Duke class.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    /**
+     * Constructor for the duke.Duke class.
+     *
+     * @param filePath The String representing the filePath of the file to be used.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            taskList = new TaskList(storage.load());
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+            taskList = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the program.
+     */
+    public void run() {
+        ui.showWelcomeMsg();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                Command command = Parser.parse(fullCommand);
+                command.execute(taskList, ui, storage);
+                isExit = command.isExit();
+            } catch (DukeException e) {
+                System.out.println(e);
+            }
+        }
+    }
+
+    public static void main(String[] args) {
```

Comment:
LGTM, I like how you formatted your code and split it into different classes.  Made it really clean and easy for reviewers to look through! 👍